homeassistant/components/withings/common.py

```diff
@@ -438,7 +438,7 @@ async def async_get_measures(self) -> dict[Measurement, Any]:
     async def async_get_sleep_summary(self) -> dict[Measurement, Any]:
         """Get the sleep summary data."""
         _LOGGER.debug("Updating withing sleep summary")
-        now = dt_util.utcnow()
+        now = dt_util.now()
         yesterday = now - datetime.timedelta(days=1)
         yesterday_noon = datetime.datetime(
```

Comment:
I suggest using `dt_util.start_of_local_day` of `yesterday` and then adding `timedelta(hours=12)` to get noon of local day. That way we keep using aware datetimes throughout.

Reply:
Makes sense! Was just making the minimal change to the existing code - implemented that fix.